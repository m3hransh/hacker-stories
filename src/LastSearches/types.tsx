
export interface LastSearchesProps {
  lastSearches: Array<string>;
  onLastSearch: (searchTerm: string) => void;
}