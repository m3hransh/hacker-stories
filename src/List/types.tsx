import {Stories, Story} from '../types';

export interface ListProps {
  list: Stories;
  onRemoveItem: (item: Story) => void;
}

export interface ItemProps {
  item: Story;
  onRemoveItem: (item: Story) => void;
}