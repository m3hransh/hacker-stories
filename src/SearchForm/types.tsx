
export interface SearchFormProps {
  searchTerm: string;
  onSearchInput: (event: React.ChangeEvent<HTMLInputElement>) => void;
  onSearchSubmit: (event: React.FormEvent<HTMLFormElement>) => void;
  className: string;
}