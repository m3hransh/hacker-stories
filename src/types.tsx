
export interface Story {
  objectID: number;
  url: string;
  title: string;
  author: string;
  num_comments: number;
  points: number;
}

export type Stories = Array<Story>;

export interface StoriesState {
  data: Stories;
  isLoading: boolean;
  isError: boolean;
  page: number;
};

interface StoriesFetchInitAction {
  type: 'STORIES_FETCH_INIT';
}

interface StoriesFetchSuccessAction {
  type: 'STORIES_FETCH_SUCCESS';
  payload: {
    list: Stories;
    page: number;
  } 
};

interface StoriesFetchFailureAction {
  type: 'STORIES_FETCH_FAILURE';
}

interface StoriesRemoveAction{
  type: 'REMOVE_STORY';
  payload: Story;
}

export type StoriesAction =
  | StoriesFetchInitAction
  | StoriesFetchSuccessAction
  | StoriesFetchFailureAction
  | StoriesRemoveAction;