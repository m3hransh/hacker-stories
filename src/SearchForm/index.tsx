
import InputWithLabel from '../InputWithLabel';
import styles from '../App.module.css';
import { SearchFormProps } from './types';

const SearchForm = ({
  searchTerm,
  onSearchInput,
  onSearchSubmit,
  className,
}: SearchFormProps) => (
  <form onSubmit={onSearchSubmit} className={className}>
    <InputWithLabel
      id="search"
      value={searchTerm}
      onInputChange={onSearchInput}
    >
      <strong>Search:</strong>
    </InputWithLabel>

    <button
      type="submit"
      disabled={!searchTerm}
      className={`${styles.button} ${styles.buttonLarge}`}
    >
      Submit
    </button>
  </form>
);

export default SearchForm;