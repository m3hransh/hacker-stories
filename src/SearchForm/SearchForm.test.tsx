import { render, screen, fireEvent, act } from "@testing-library/react";
import {SearchFormProps} from './types';
import SearchForm from './index';

describe("SearchForm", () => {
  const searchFormProps: SearchFormProps = {
    searchTerm: "React",
    onSearchInput: jest.fn(),
    onSearchSubmit: jest.fn(),
    className: "Search",
  };

  test("renders the input field with its value", () => {
    render(<SearchForm {...searchFormProps} />);

    expect(screen.getByDisplayValue('React')).toBeInTheDocument();
  });

  test("renders the correct label", () => {
    render(<SearchForm {...searchFormProps} />);

    expect(screen.getByLabelText(/Search/)).toBeInTheDocument();
  });

  test("calls onSearchInput on input field change", () => {
    render(<SearchForm {...searchFormProps} />);

    fireEvent.change(screen.getByDisplayValue('React'), {
      target: { value: 'Redux' },
    });

    expect(searchFormProps.onSearchInput).toHaveBeenCalledTimes(1);
  });

  test("calls onSearchSubmit on button submit click", () => {
    render(<SearchForm {...searchFormProps} />);

    fireEvent.submit(screen.getByRole('button'));

    expect(searchFormProps.onSearchSubmit).toHaveBeenCalledTimes(1);
  });

  test('renders snapshot', () => {
    const { container } = render(<SearchForm {...searchFormProps} />);
    expect(container.firstChild).toMatchSnapshot();
  });
});