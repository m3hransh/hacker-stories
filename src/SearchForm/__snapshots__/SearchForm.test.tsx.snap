// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchForm renders snapshot 1`] = `
<form
  class="Search"
>
  <label
    class="label"
    for="search"
  >
    <strong>
      Search:
    </strong>
  </label>
  Â 
  <input
    class="input"
    id="search"
    type="text"
    value="React"
  />
  <button
    class="button buttonLarge"
    type="submit"
  >
    Submit
  </button>
</form>
`;
